tests/framework/e2e/cluster.go

```diff
@@ -182,6 +182,9 @@ type EtcdProcessClusterConfig struct {
 	CorruptCheckTime        time.Duration
 	CompactHashCheckEnabled bool
 	CompactHashCheckTime    time.Duration
+
+	UserName string // user/password pair to be used for client connections
+	Password string
```

Comment:
There should be no need to add these two fields. Please refer to https://github.com/etcd-io/etcd/blob/e24402d39fd08ebca46beeb1f16d214e37fea74f/tests/framework/e2e.go#L93

Reply:
The problem I have is StartNewProc() calls Client() underneath which fails if Auth is enabled.
https://github.com/etcd-io/etcd/blob/e24402d39fd08ebca46beeb1f16d214e37fea74f/tests/framework/e2e/cluster.go#L527-L532
Another option would be to pass credentials to StartNewProc() and likely other methods directly, but that doesn't look like a good solution to me.